src/jaxsim/rbda/contacts/relaxed_rigid.py

```diff
@@ -155,18 +162,19 @@ class RelaxedRigidContactsState(ContactsState):
     def __eq__(self, other: RelaxedRigidContactsState) -> bool:
         return hash(self) == hash(other)
 
-    @staticmethod
-    def build() -> RelaxedRigidContactsState:
+    @classmethod
+    def build(cls: type[Self]) -> Self:
         """Create a `RelaxedRigidContactsState` instance"""
 
         return RelaxedRigidContactsState()
 
-    @staticmethod
-    def zero(model: js.model.JaxSimModel) -> RelaxedRigidContactsState:
+    @classmethod
+    def zero(cls: type[Self], *, model: js.model.JaxSimModel) -> Self:
```

Comment:
We can actually remove `model` since it is not used. The signature shouldn't change with respect to the abstract class
```suggestion
    def zero(cls: type[Self]) -> Self:
```